File: sms-api-java/userservice/src/main/java/com/example/mydemoproject/userservice/controller/UserController.java
package com.example.mydemoproject.userservice.controller;

import com.example.mydemoproject.userservice.dto.UserDto;
import com.example.mydemoproject.userservice.model.User;
import com.example.mydemoproject.userservice.service.UserService;
import jakarta.ws.rs.QueryParam;
import lombok.AllArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("api/users")
@AllArgsConstructor
public class UserController {

    private UserService userService;

    @PostMapping
    public ResponseEntity<User> saveUser(@RequestBody User user) {
        User savedUser = userService.saveUser(user);
        return new ResponseEntity<>(savedUser, HttpStatus.CREATED);
    }

    @GetMapping("{id}")
    public ResponseEntity<UserDto> getUser(@PathVariable("id") Long userId) {
        UserDto userDto = userService.getUser(userId);
        return ResponseEntity.ok(userDto);
    }

//    @GetMapping
//    public ResponseEntity<UserDto> findUserByEmail(@QueryParam("email") String email) {
//        UserDto userDto = userService.findUserByEmail(email);
//        return ResponseEntity.ok(userDto);
//    }
}

File: sms-api-kotlin/userservice/src/main/kotlin/com/example/userservice/controller/UserController.kt
package com.example.userservice.controller

import com.example.userservice.controller.dto.ApiResponse
import com.example.userservice.controller.dto.UserDto
import com.example.userservice.model.User
import com.example.userservice.model.toUserDto
import com.example.userservice.service.UserServiceImpl
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("v1/users")
class UserController (val userServiceImpl: UserServiceImpl){

    var apiResponse: ApiResponse? = null

    @GetMapping
    fun getAllUsers() : List<UserDto>{
        var users = userServiceImpl.getAllUsers()
        return users.map { it.toUserDto() }
    }

    @GetMapping("/{id}")
    fun getUser(@PathVariable id: Long): ResponseEntity<UserDto>{
        var user = userServiceImpl.getUserById(id)
        var userDto = user.toUserDto()
        return ResponseEntity.ok().body(userDto)
    }

    @PostMapping("/add")
    fun saveUser(@RequestBody user: User) : ResponseEntity<ApiResponse> {
        var savedUser = userServiceImpl.saveUser(user)
        savedUser.toUserDto()
        apiResponse = ApiResponse(
            "8",
            "User with username ${savedUser.username} successfully created.",
            "https://oziz.ffos.hr/nastava20192020/mmilosevic_19/DiplomskiRad/HTML-API-Docs/users/v1/index.html"
        )
        return ResponseEntity.status(HttpStatus.OK).body(apiResponse)
    }

    @DeleteMapping("/{id}")
//    @ResponseStatus(HttpStatus.OK)
    fun deleteUser(@PathVariable id: Long) : ResponseEntity<ApiResponse>{
        userServiceImpl.deleteUser(id)
        apiResponse = ApiResponse(
            "11",
            "Deleted",
            "https://oziz.ffos.hr/nastava20192020/mmilosevic_19/DiplomskiRad/HTML-API-Docs/users/v1/index.html"
        )
        return ResponseEntity.status(HttpStatus.OK).body(apiResponse)
    }

    @PutMapping("/{idUser}")
    fun updateUser(@PathVariable idUser: Long, @RequestBody user: User): ResponseEntity<ApiResponse> {
        var updatedUser = userServiceImpl.updateUser(idUser, user)
        updatedUser.toUserDto()
        apiResponse = ApiResponse(
            "9",
            "User with username ${updatedUser.username} successfully updated.",
            "https://oziz.ffos.hr/nastava20192020/mmilosevic_19/DiplomskiRad/HTML-API-Docs/users/v1/index.html"
        )
        return ResponseEntity.status(HttpStatus.OK).body(apiResponse)
    }
}
